packages/neon-api/src/transaction/validator.ts

```diff
@@ -0,0 +1,229 @@
+   */
+  validateValidUntilBlock = async (
+    autoFix = false
+  ): Promise<ValidationResult> => {
+    const { validUntilBlock } = this.transaction;
+    const height = await this.rpcClient.getBlockCount();
+    if (
+      validUntilBlock <= height ||
+      validUntilBlock >= height + tx.Transaction.MAX_TRANSACTION_LIFESPAN
+    ) {
+      const vub_suggestion =
+        tx.Transaction.MAX_TRANSACTION_LIFESPAN + height - 1;
+      if (autoFix) {
+        this.transaction.validUntilBlock = vub_suggestion;
+        return {
+          valid: true,
+          validaUntilBlock: {
+            fixed: true,
+            prev: validUntilBlock,
+            suggestion: vub_suggestion
+          }
+        };
+      }
+      return {
+        valid: false,
+        validaUntilBlock: {
+          fixed: false,
+          prev: validUntilBlock,
+          suggestion: vub_suggestion
+        }
+      };
+    }
+    return {
+      valid: true
+    };
+  };
+
+  // validateIntents = async (): Promise<ValidationResult> => {
+  //   return {
+  //     result: true,
+  //     fixed: false
+  //   };
+  // };
+
+  /**
+   * validate systemFee
+   * @param autoFix will automatically fix transaction if specified as true
+   */
+  validateSystemFee = async (autoFix = false): Promise<ValidationResult> => {
+    const { script, systemFee } = this.transaction;
+    const { gas_consumed } = await this.rpcClient.invokeScript(script);
+    const minimumSystemFee = new u.Fixed8(parseFloat(gas_consumed)).ceil();
+    if (autoFix && !minimumSystemFee.equals(systemFee)) {
+      this.transaction.systemFee = minimumSystemFee;
+      return {
+        valid: true,
+        systemFee: {
+          fixed: true,
+          prev: systemFee,
+          suggestion: minimumSystemFee
+        }
+      };
+    } else if (minimumSystemFee.isGreaterThan(systemFee)) {
+      return {
+        valid: false,
+        systemFee: {
+          fixed: false,
+          prev: systemFee,
+          suggestion: minimumSystemFee
+        }
+      };
+    }
+    return {
+      valid: true
+    };
+  };
+
+  /**
+   * Validate NetworkFee
+   * @param autoFix will automatically fix transaction if specified as true
+   */
+  validateNetworkFee = async (autoFix = false): Promise<ValidationResult> => {
+    const { networkFee } = this.transaction;
+    const minimumNetworkFee = getNetworkFee(this.transaction);
+    if (autoFix && !minimumNetworkFee.equals(networkFee)) {
+      this.transaction.networkFee = minimumNetworkFee;
+      return {
+        valid: true,
+        networkFee: {
+          fixed: true,
+          prev: networkFee,
+          suggestion: minimumNetworkFee
+        }
+      };
+    } else if (minimumNetworkFee.isGreaterThan(networkFee)) {
+      return {
+        valid: false,
+        networkFee: {
+          fixed: false,
+          prev: networkFee,
+          suggestion: minimumNetworkFee
+        }
+      };
+    }
+    return {
+      valid: true
```

Comment:
So there isnt a suggestion if we do not want autofix but attached a larger than needed fee?

Reply:
Yes. Anyway validator is mostly to validate, to save money is secondary job. But it's harmless if we add a check here, what do you think?

Reply:
If you say that, then autofix should not fix if the attached fee is larger than required.

Reply:
Well I think autoFix could fix when attached fee is larger than required.
It will not give suggestions when autoFix = false and attached fee is larger than required.
What do you think?

Reply:
I think most people will expect the `autoFix` flag to just apply the suggestion and not change the suggestion output based on that flag. Please make it consistent.

Reply:
OK.